### PR TITLE
Remove remaining binary punch diagnostics code

### LINKED ISSUE #82: [FEATURE REQUEST] Removal of binary punch diagnostics
## Overview ##
We are removing the legacy binary punch (aka bpch) diagnostics, because they cannot be used by GCHP, as well as GEOS-Chem connected to other ESMs (GEOS, WRF, CESM, etc).

## Validation ##
Prior to removing the bpch diagnostics, we performed several validations:

### Comparison of bpch and netCDF diagnostic output ###
Please see our [Validation of netCDF diagnostics section](http://wiki.geos-chem.org/Benchmark/GEOS-Chem_12.5.0#Validation_of_netCDF_diagnostics) on the GEOS-Chem wiki.

### Difference tests ###
```
Ref = bpch code present
Dev = bpch code removed (feature/RemoveBpch branch)

Compiled with BPCH_DIAG=n (this will be the new default)
---------------------------------------------------------
DiffTest_geosfp_2x25_CO2   	        PASS
DiffTest_geosfp_4x5_aerosol/            PASS	
DiffTest_geosfp_4x5_benchmark/          PASS
DiffTest_geosfp_4x5_CH4/                PASS	
DiffTest_geosfp_4x5_complexSOA/         PASS
DiffTest_geosfp_4x5_standard/           PASS
DiffTest_geosfp_4x5_tagCO/              PASS
DiffTest_geosfp_4x5_tagO3/              PASS			
DiffTest_geosfp_4x5_TransportTracers/   PASS
DiffTest_geosfp_4x5_tropchem/           PASS

Compiled with BPCH_DIAG=y
---------------------------------------------------------
DiffTest_geosfp_2x25_CO2/               PASS
DiffTest_geosfp_4x5_aerosol/		PASS
DiffTest_geosfp_4x5_benchmark/          PASS**
DiffTest_geosfp_4x5_CH4/                PASS
DiffTest_geosfp_4x5_Hg/                 PASS          
DiffTest_geosfp_4x5_standard/           PASS*
DiffTest_geosfp_4x5_tagCO/              PASS
DiffTest_geosfp_4x5_tagO3/	        PASS			
DiffTest_geosfp_4x5_TransportTracers/   PASS
DiffTest_geosfp_4x5_tropchem/           PASS**

*: The perl script validate.pl, which is used to check for differences, 
   notes that certain diagnostic outputs are different.
   But when examined in Python, we get identical results.  
   This might denote some very small differences caused by optimization.   
   For all intents and purposes this denotes dentical results.

**: Differences are within the limits of numerical noise.

The Mean OH value for Dev in geosfp_4x5_standard is identical with bpch turned off or on:

Bpch off : Mean OH =    11.551113072639739       [1e5 molec/cm3]
Bpch on  : Mean OH =    11.551113072639739       [1e5 molec/cm3]

as is the mean OH value for Dev in geosfp_4x5_tropchem:

Bpch off : Mean OH =    12.979569056826467       [1e5 molec/cm3]
Bpch on  : Mean OH =    12.979569056826467       [1e5 molec/cm3]

and as is the mean OH value for Dev in geosfp_4x5_benchmark:

Bpch off : Mean OH =    11.544115278980730       [1e5 molec/cm3]
Bpch on  : Mean OH =    11.544115278980730       [1e5 molec/cm3]
```

### Unit Tests ###
ut

## Remaining bpch diagnostics ##
Due to legacy code in certain "specialty" simulations, we are unable to remove every single bpch diagnostic.  For the time being we will preserve the following bpch diagnostics:

1. ND03 (for Hg simulations)
2. ND06 (for TOMAS simulations)
3. ND44 (for TOMAS simulations)
4. ND51 and ND51b (Satellite timeseries)
5. ND53 (for POPs simulations)
6. ND59 (for TOMAS simulations)
7. ND60 (for TOMAS simulations)
8. ND61 (for TOMAS simulations)
9. ND72 (for RRTMG simulations)

### PR DESCRIPTION
**Pull request by yantosca**
### Name and Institution (Required)

Name: Bob Yantosca
Institution: Harvard + GCST

### Confirm you have reviewed the following documentation

- [x] [Contributing guidelines](https://geos-chem.readthedocs.io/en/stable/help-and-reference/CONTRIBUTING.html)

### Describe the update

This PR is the companion to #82. 

We have removed the following binary punch diagnostics files:

- GeosCore/diag3.F90
- GeosCore/diag_mod.F90
- GeosCore/gamap_mod.F90
- GeosCore/initialize.F90
- GeosCore/ndxx_setup.F90
- GeosUtil/bpch2_mod.F90
- Headers/CMN_DIAG_mod.F90
- Headers/CMN_O3_mod.F90

We have removed references to the above files from the following CMakeLists:

- Headers/CMakeLists.txt
- GeosCore/CMakeLists.txt
- GeosUtil/CMakeLists.txt

We have removed code contained inside `#ifdef BPCH_DIAG` blocks in these modules:

- GeosCore/aero_drydep.F90
- GeosCore/carbon_mod.F90
- GeosCore/chemistry_mod.F90
- GeosCore/dust_mod.F90
- GeosCore/flexgrid_read_mod.F90
- GeosCore/gc_environment_mod.F90
- GeosCore/hcoi_gc_diagn_mod.F90
- GeosCore/input_mod.F90
- GeosCore/ocean_mercury_mod.F90
- GeosCore/planeflight_mod.F90
- GeosCore/sulfate_mod.F90
- GeosCore/tomas_mod.F90
- GeosUtil/time_mod.F90
- Headers/input_opt_mod.F90
- Interfaces/GCClassic/main.F90

And we have removed references to bpch diagnostics in the following configuration file templates:

- run/GCClassic/geoschem.config.yml.templates/geoschem_config.yml 
- run/WRF/ch4/geoschem_config.yml
- run/WRF/fullchem/geoschem_config.yml
- Removed the "legacy_bpch" YAML section and subsections

as well as in the test/shared/commonFunctionsForTests.sh integration test script.

### Expected changes

This is a zero-diff update.  However, we should merge this into the dev/14.4.0 branch as it touches run directory configuration files.

### Related Github Issue(s)

- Closes #82
- Merge before https://github.com/geoschem/GCClassic/pull/49

### CODE REVIEW

File: GeosCore/cleanup.F90 [hunk 1]
```diff
@@ -201,25 +197,6 @@ SUBROUTINE CLEANUP( Input_Opt, State_Grid, ERROR, RC )
...
-
-#ifdef RRTMG
-  CALL Cleanup_RRTMG_Rad_Transfer( RC )
-  IF ( RC /= GC_SUCCESS ) THEN
-     ErrMsg = 'Error encountered in "Cleanup_RRTMG_Rad_Transfer"!'
-     CALL GC_Error( ErrMsg, RC, ThisLoc )
-     RETURN
-  ENDIF
-#endif
-
```
Comment (msulprizio) [inline GeosCore/cleanup.F90:222]: I believe this was inadvertently removed? Please restore if so.

File: GeosCore/hcoi_gc_diagn_mod.F90 [hunk 2]
```diff
@@ -677,7 +665,7 @@ SUBROUTINE Diagn_CH4( Input_Opt, HcoState, ExtState, RC )
 
     IF ( HcoID > 0 ) THEN
 
-       ! Create diagnostic container
+       ! Create diagnostic container<
```
Comment (msulprizio) [inline GeosCore/hcoi_gc_diagn_mod.F90:668]: Remove extraneous "<" at end of line

**Review by msulprizio — CHANGES_REQUESTED** [review]

**Review by msulprizio — APPROVED** [review]
Both minor requested changes have been addressed in https://github.com/geoschem/geos-chem/pull/2166/commits/96b4c7d65fa61adee50d392c53c3604636f99126. Good to merge.

**Comment by yantosca** [comment]
This PR has been merged (locally) atop PR #1998, and integration tests are running.

**Comment by yantosca** [comment]
All GEOS-Chem Classic integration tests passed:
==============================================================================
GEOS-Chem Classic: Execution Test Results

GCClassic #d51e872 Remove BPCH_DIAG compilation option and related documentation
GEOS-Chem #7ab17802a Merge PR #2166 (Remove remaining binary punch diagnostics code)
HEMCO     #f807e1a Update HEMCO version number to 3.8.0 in preparation for release

Using 24 OpenMP threads
Number of execution tests: 26

Submitted as SLURM job: 21856850
==============================================================================


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```

**Comment by yantosca** [comment]
All GCHP integration tests also passed:
```console
==============================================================================
GCHP: Execution Test Results

GCHP      #3dbf81e GEOS-Chem submod update: Merge PR #1998 (Rename YAML tags)
GEOS-Chem #b0bc778d7 PR #2166 post-merge fix: Remove #ifdef block in read_input_file
HEMCO     #f807e1a Update HEMCO version number to 3.8.0 in preparation for release

Number of execution tests: 6

Submitted as SLURM job: 21863342
==============================================================================

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```

**Pull request merged**